leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/NoValueConverter.java

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Sierra Wireless and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *     Sierra Wireless - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.leshan.core.node.codec;
+
+import org.eclipse.leshan.core.model.ResourceModel.Type;
+import org.eclipse.leshan.core.node.LwM2mPath;
+
+/**
+ * A {@link LwM2mValueConverter} which do no conversion but raise a {@link CodecException} if type is not the expected
+ * one.
+ *
+ */
+public class NoValueConverter implements LwM2mValueConverter {
```

Comment:
This `NoValue` name is bit confusing. It is more a `NoConversionConverter` 😄.
`CheckTypeConverter`?

Reply:
I renamed it by `LwM2mValueChecker`